pkg/engine/planner/physical/planner.go

```diff
@@ -363,8 +365,29 @@ func (p *Planner) processVectorAggregation(lp *logical.VectorAggregation, ctx *C
 	return []Node{node}, nil
 }
 
-// Optimize tries to optimize the plan by pushing down filter predicates and limits
-// to the scan nodes.
+// Convert [logical.Parse] into one [ParseNode] node.
+func (p *Planner) processParse(lp *logical.Parse, ctx *Context) ([]Node, error) {
+	node := &ParseNode{
```

Comment:
nit: it may be worth adding a comment explaining that RequestedKeys gets filled during an optimization pass